day_09.md

```diff
@@ -0,0 +1,265 @@
+        console.log(`D`);
+      }, 1000 );
+    }, 100 );
+  }, 0 );
+}, 5000 );
+
+```
+
+After a few moments thinking about it you might say "that's easy", I won't gent into the "using `console`" implications, but let's imagine that you need `B`to happen only if `A`  is successful, AND `C`AND `D` should happen either way. Now our code starts looking not only hard to read but also hard to maintain, which is the doorway to a tsunami of bugs swarming to you.
+
+So, what's a promise and why it might help here?
+
+Imagine this:
+> I'm an Object, and once you give me a specific task I PROMISE YOU I'll let you know when I did it,  telling wether I SUCCEEDED or FAILED providing you the summary (outcome) so you can use it, THEN you can do something with that info if I SUCCEEDED OR alternatively do something with my FAILURE. FINALLY you might want to do something either way.
+
+That's something you are used to do with other people, isn't it?
+
+#### Syntax
+
+```javascript
+  new Promise(executor);
+  // or
+  new Promise((resolve, reject) => {
+    executor body
+  });
+```
+
+Well, Promises work kinda that.
+
+```javascript
+  new Promise((resolve, reject) => {
+      console.log('I started');
+      resolve();
+  })
+  .then(() => {
+      console.log(`A`);
+      throw new Error('I failed in A');
+  })
+  .then(() => {
+      console.log(`B`);
+  })
+  .catch((reason) => {
+      console.log(reason);
+  })
+  .finally(() => {
+      console.log(`C`);
+      console.log(`D`);
+  });
+
+```
+
+Easy to read, easy to control ... sort of, but definitely better than callback hell.
+
+Too much talking, let go to the best sources you can get.
+
+- [ECMAScript Promise Objects](https://www.ecma-international.org/ecma-262/6.0/#sec-promise-objects)
+- [ECMAScript Promise Constructor](https://www.ecma-international.org/ecma-262/6.0/#sec-promise-constructor)
+- [MDN - Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise)
+- [MDN - Using Promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises)
+- [YDKJS - Async & Performance - Chapter 3: Promises](https://github.com/getify/You-Dont-Know-JS/blob/1st-ed/async%20%26%20performance/ch3.md) | by Kyle Simpson
+
+---
+
+### Async/await
+
+Wait what? we were talking about ES2015 but this is not an ES2015 feature, nor ES2016!! We had to wait until ES2017 (ES8?? nope, remember that's not the way to name ES versions anymore) to have it.
+Then why are we talking about this? Simply because is one of the most important tools (together with generators and promises) to write asynchronous programs in JavaScript today.
+
+Imagine there's a feature to handle under the hood a the event loop and the promises to **operate asynchronously BUT resembles synchronous** code on its syntax. Since many program errors are originated on the difficulty of understanding a complex structure (e.g. like a promise chain) this feature might help with that, isn't it?
```

Comment:
Could the "a" be removed? "handle under the hood **a** the event loop and the promises"